### ISSUE TITLE
Support selection and copy of all ontology metrics in the "Ontology metrics" view

### ISSUE THREAD

**Issue by donpellegrino** [protegeproject/protege#598]
It would be a useful enhancement to be able to select all content from the "Ontology metrics" view so that it could be pasted into other applications within a graphical desktop environment. Within each section, it seems that a table is used. Selection can be done for each row to activate a highlight however the row content does not get added to the clipboard with CTRL+C. Perhaps this is an artifact of the Java widget toolkit? In general, it would be useful if all content within the application was selectable and copyable.

Test system:

------------------------------------ Protege ----------------------------------- 
Protege Desktop 
Version 5.1.0, Build  
----------------------------------- Platform ----------------------------------- 
Java: JVM 1.8.0_111-b15  Memory: 466M 
Language: en, Country: US 
Framework: Apache Software Foundation (1.7)  
OS: linux (3.10.0-514.2.2.el7.x86_64) 
Processor: x86-64

$ java -version
openjdk version "1.8.0_111"
OpenJDK Runtime Environment (build 1.8.0_111-b15)
OpenJDK 64-Bit Server VM (build 25.111-b15, mixed mode)

**Comment by csnyulas** [comment]
Hi Don,

This works for me fine (note: you can export metrics from one section at 
a time).
I created a video that demonstrates this:
https://www.youtube.com/watch?v=wLWnYsnBYgI

If you are not able to do exactly what it is shown in the video, there 
may be an issue with OpenJDK. Can you try to run Protege with the Oracle 
Java, and see if you still encounter problems?

Csongor

**Comment by donpellegrino** [comment]
@csnyulas Thanks for putting the video together. That helps get to the detail nicely. I am able to reproduce the behavior from the video with OpenJDK. I did not test with Oracle Java. At 23 seconds of the video observe that "DL expressivity" under Metrics was selected for the CTRL+C operation. The subsequent paste operation failed to paste that content.

A work-around is to manually select each table in the Metrics one at a time and then paste them one at a time. However, it would be a convenience if content from multiple sections or all sections could be selected and copied together.

**Comment by csnyulas** [comment]
Yes, your observation (which I have also noted in a parenthesis) is correct: you can copy metrics from one section at a time, as they are in separate JTables and the CTRL+C works on only one JTable (the active one) at a time. I don't think it would be an easy fix (if at all possible) to make Protege do what you want with the current organization of the information. We would probably need to rewrite the Metrics view in order to achieve what you are asking for. I think the work around that both you and I found (to copy out the content of each section separately) is not that bad, and probably does not warrant the effort that would be needed to rewrite the Metrics view. Maybe @matthewhorridge can chime in on this.

**Comment by donpellegrino** [comment]
@csnyulas Fair enough, if significant effort is needed for the change then it it probably not worth it. Copying content to the clipboard from multiple JTables at once might be more appropriate as an enhancement or bug report against the underlying Java Runtime if that is where the behavior is coming from.
